python/elastic_search_serialization.py

```diff
@@ -0,0 +1,474 @@
+        raise_on_error (bool): Raise BulkIndexError containing errors (as .errors) from the execution of the last chunk when some occur. By default we raise.
+        raise_on_exception (bool): If False then don’t propagate exceptions from call to bulk and just report the items that failed as failed.
+        max_retries (int): Maximum number of times a document will be retried when 429 is received, set to 0 (default) for no retries on 429.
+        initial_backoff (float): The number of seconds we should wait before the first retry. Any subsequent retries will be powers of initial_backoff * 2**retry_number.
+        max_backoff (float): The maximum number of seconds a retry will wait.
+        yield_ok (float): If set to False will skip successful documents in the output.
+    """
+    for _, _ in streaming_bulk(
+        client=client,
+        index=index,
+        actions=objects,
+        chunk_size=chunk_size,
+        max_chunk_bytes=max_chunk_bytes,
+        initial_backoff=initial_backoff,
+        max_backoff=max_backoff,
+        yield_ok=yield_ok,
+        raise_on_error=raise_on_error,
+        raise_on_exception=raise_on_exception,
+        max_retries=max_retries,
+    ):
+        pass
+
+
+@mgp.read_proc
+def connect(
+    elastic_url: str, ca_certs: str, elastic_user: str, elastic_password
+) -> mgp.Record(connection_status=mgp.Map):
+    """Establishes connection with the Elasticsearch. This configuration needs to be specific to the Elasticsearch deployment. Uses basic authentication
+    Args:
+        elastic_url (str): URL for connecting to the Elasticsearch instance.
+        ca_certs (str): Path to the certificate file.
+        elastic_user (str): The user trying to connect to the Elasticsearch.
+        elastic_password (str): User's password for connecting to the Elasticsearch.
+    Returns:
+        mgp.Record(connection_status=mgp.Map): Connection info.
+    """
+    global client
+    client = elasticsearch.Elasticsearch(
+        hosts=elastic_url,
+        ca_certs=ca_certs,
+        basic_auth=(elastic_user, elastic_password),
+    )
+    logger.info(f"Client info: {client.info()}")
+    return mgp.Record(connection_status=dict(client.info()))
+
+
+@mgp.read_proc
+def create_index(
+    context: mgp.ProcCtx,
+    index_name: str,
+    schema_path: str,
+    schema_parameters: mgp.Map,
+) -> mgp.Record(response=mgp.Map):
+    """Creates index with the given index name.
+    Args:
+        index_name (str): Name of the index that needs to be created.
+        schema_path (str): Path to the schema from where it will be loaded.
+        schema_parameters: Dict[str, Any]
+            number_of_shards (int): Number of shards index will use.
+            number_of_replicas (int): Number of replicas index will use.
+        analyzer (str): Custom analyzer, can be set to any legal Elasticsearch analyzer.
+    Returns:
+       mgp.Map: Response message from Elasticsearch service.
+    """
+    global client
```

Comment:
Maybe one idea here. Instead of using a `global client`, we could create a `Singleton` class and here get a `client` from that class. And in the method connect, you can initialize that class. This way we could provide unit tests only, but it would be enough to check that the implementation works, right?